DiceKeys/Core/Api/Unmarshalling.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  Unmarshalling.swift
+//  DiceKeys
+//
+//  Created by Stuart Schechter on 2021/02/04.
+//
+
+import Foundation
+
+protocol ApiRequestParameterUnmarshaller {
+    func optionalField(name fieldName: String) -> String?
+    func requiredField(name fieldName: String) throws -> String
+}
+
+class UrlParameterUnmarshaller: ApiRequestParameterUnmarshaller {
+    private let parameters: [String: String?]
+
+    init(url: URL) {
+        var queryDictionary = [String: String?]()
```

Comment:
I would suggest making the dictionary value non-optional [String: String]. Maintaining nil values in memory is just overhead. If a key is not found in dictionary then it returns nil anyway

Reply:
Fixing